analyses/molecular-subtyping-HGG/01-HGG-molecular-subtyping-defining-lesions.Rmd

```diff
@@ -85,12 +90,22 @@ snv_lesions_df <- snv_df %>%
   dplyr::select(
     -HGVSp_Short,
     -Hugo_Symbol
-  )
+  ) 
+
+# add back in samples with no evidence of these specific mutations
+snv_lesions_df <- snv_lesions_df %>%
+  dplyr::bind_rows(
+    data.frame(
+      Tumor_Sample_Barcode = setdiff(unique(snv_df$Tumor_Sample_Barcode),
+                                     snv_lesions_df$Tumor_Sample_Barcode)
+    )
+  ) %>%
+  dplyr::mutate_all(function(x) tidyr::replace_na(x, "No"))
 
 # Join the selected variables from the metadata with the snv consensus mutation
 # and defining lesions data.frame
 snv_lesions_df <- select_metadata %>%
-  dplyr::right_join(snv_lesions_df,
+  dplyr::inner_join(snv_lesions_df,
```

Comment:
This change has to do with the changes at lines 46-48. Before, we had no filter to just tumor samples and the histologies file has samples from multiple experimental strategies. Now we want to remove cell line data (which have been dropped out of `metadata` and therefore `select_metadata`) which would not be accomplished with a right join and we can't do a left join because `select_metadata` will have RNA-seq specimens. Do you agree with my logic?

Reply:
Okay. Yes this sounds good. Was a bit tricky to follow these changes and how they were connected from a glance but this makes sense.

Reply:
They were tricky to write about, too 😅